File: 2262-solving-questions-with-brainpower/2262-solving-questions-with-brainpower.cpp
class Solution {
public:
    long long mostPoints(vector<vector<int>>& questions) {
        int n=questions.size();
        long long dp[n+1];
        memset(dp,0,sizeof(dp));
       
        for(int i=n-1;i>=0;i--){
           int points = questions[i][0], jump = questions[i][1];
           dp[i]=max(dp[i+1],points+dp[min(i+1+jump,n)]);
        }
        return dp[0];
    }
};

File: 2262-solving-questions-with-brainpower/2262-solving-questions-with-brainpower.js
/**
 * @param {number[][]} questions
 * @return {number}
 */
var mostPoints = function (questions) {
    const n = questions.length;
    const dp = new Array(n + 1).fill(0);

    for (let i = n - 1; i >= 0; i--) {
        const points = questions[i][0];
        const skip = questions[i][1];

        const nextIndex = i + skip + 1;
        dp[i] = Math.max(points + (nextIndex < n ? dp[nextIndex] : 0), dp[i + 1]);
    }

    return dp[0];
};